flux-desugar/src/table_resolver.rs

```diff
@@ -462,33 +500,42 @@ impl<'sess> NameResTable<'sess> {
                 };
                 self.collect_from_path(path)
             }
+            hir::TyKind::OpaqueDef(item_id, args, _) => {
+                self.opaque = Some((item_id.owner_id.def_id, args.len()));
```

Comment:
We should report an error or at least panic here if `self.opaque` is already `Some`.